common/src/main/scala/uk/ac/wellcome/utils/TryBackoff.scala

```diff
@@ -67,13 +67,13 @@ trait TryBackoff extends Logging {
         right => 0
       )
 
-      if (numberOfAttempts > maxAttempts) {
-        throw new RuntimeException("Max retry attempts exceeded")
-      }
-
-      val waitTime = timeToWaitOnAttempt(attempt)
+      val shouldReschedule = if (numberOfAttempts > maxAttempts) {
```

Comment:
Not for here, but worth thinking about: what would this look like if we had a "reschedule" function that returned an Option of Cancellable? Could we push some of this conditional logic into the `fold` above?